### PR TITLE
Enable functionality for certificate based authentication

### PR DESCRIPTION
**Pull request by arudell**
# Description
Summary of changes:
- Enabled x509 authentication to be used when validating against REST endpoint for NC API: `-NcRestCertificate`
- Standardized all functions that interact with REST endpoint to use the following parameter names:
    - `NcRestCredential`
    - `NcRestCertificate`
- Fixed logic handling for several functions
- Fixed issue with determining FCNC clusters

# Change type
- [ ] Bug fix (non-breaking change)
- [ ] Code style update (formatting, local variables)
- [x] New Feature (non-breaking change that adds new functionality without impacting existing)
- [x] Breaking change (fix or feature that may cause functionality impact)
- [ ] Other

# Checklist:
- [x] My code follows the style and contribution guidelines of this project.
- [x] I have tested and validated my code changes.

### CODE REVIEW

File: src/modules/SdnDiag.Common/public/Enable-SdnVipTrace.ps1 [hunk 1]
```diff
@@ -30,47 +33,64 @@ function Enable-SdnVipTrace {
...
+        NcUri = $NcUri
+        ErrorAction = 'Stop'
+    }
+
+    switch ($PSCmdlet.ParameterSetName) {
+        'RestCertificate' {
+            $ncRestParams.Add('Certificate', $NcRestCertificate)
+        }
+        'RestCredential' {
+            $ncRestParams.Add('Credential', $NcRestCredential)
```
Comment (luyaof) [inline src/modules/SdnDiag.Common/public/Enable-SdnVipTrace.ps1:69]: Should be `NcRestCertificate` and `NcRestCredential` ?

File: src/modules/SdnDiag.Health/private/Test-MuxBgpConnectionState.ps1 [hunk 2]
```diff
@@ -9,12 +9,27 @@ function Test-MuxBgpConnectionState {
...
         $Credential = [System.Management.Automation.PSCredential]::Empty,
 
-        [Parameter(Mandatory = $false)]
+        [Parameter(Mandatory = $false, ParameterSetName = 'RestCredential')]
         [System.Management.Automation.PSCredential]
         [System.Management.Automation.Credential()]
-        $NcRestCredential = [System.Management.Automation.PSCredential]::Empty
+        $NcRestCredential = [System.Management.Automation.PSCredential]::Empty,
+
+        [Parameter(Mandatory = $true, ParameterSetName = 'RestCertificate')]
```
Comment (luyaof) [inline src/modules/SdnDiag.Health/private/Test-MuxBgpConnectionState.ps1:17]: `RestCertificate` should not be Mandatory?

File: src/modules/SdnDiag.Health/private/Test-NcHostAgentConnectionToApiService.ps1 [hunk 3]
```diff
@@ -14,12 +14,27 @@ function Test-NcHostAgentConnectionToApiService {
...
         $Credential = [System.Management.Automation.PSCredential]::Empty,
 
-        [Parameter(Mandatory = $false)]
+        [Parameter(Mandatory = $false, ParameterSetName = 'RestCredential')]
         [System.Management.Automation.PSCredential]
         [System.Management.Automation.Credential()]
-        $NcRestCredential = [System.Management.Automation.PSCredential]::Empty
+        $NcRestCredential = [System.Management.Automation.PSCredential]::Empty,
+
+        [Parameter(Mandatory = $true, ParameterSetName = 'RestCertificate')]
```
Comment (luyaof) [inline src/modules/SdnDiag.Health/private/Test-NcHostAgentConnectionToApiService.ps1:22]: Review `Mandatory` attribute

File: src/modules/SdnDiag.Health/private/Test-NetworkControllerCertCredential.ps1 [hunk 4]
```diff
@@ -14,20 +14,35 @@ function Test-NetworkControllerCertCredential {
...
         $Credential = [System.Management.Automation.PSCredential]::Empty,
 
-        [Parameter(Mandatory = $false)]
+        [Parameter(Mandatory = $false, ParameterSetName = 'RestCredential')]
         [System.Management.Automation.PSCredential]
         [System.Management.Automation.Credential()]
-        $NcRestCredential = [System.Management.Automation.PSCredential]::Empty
+        $NcRestCredential = [System.Management.Automation.PSCredential]::Empty,
+
+        [Parameter(Mandatory = $true, ParameterSetName = 'RestCertificate')]
```
Comment (luyaof) [inline src/modules/SdnDiag.Health/private/Test-NetworkControllerCertCredential.ps1:22]: Review `Mandatory` attribute

File: src/modules/SdnDiag.Health/private/Test-NetworkInterfaceAPIDuplicateMacAddress.ps1 [hunk 5]
```diff
@@ -4,24 +4,39 @@ function Test-NetworkInterfaceAPIDuplicateMacAddress {
...
         [SdnFabricEnvObject]$SdnEnvironmentObject,
 
-        [Parameter(Mandatory = $false)]
+        [Parameter(Mandatory = $false, ParameterSetName = 'RestCredential')]
         [System.Management.Automation.PSCredential]
         [System.Management.Automation.Credential()]
-        $NcRestCredential = [System.Management.Automation.PSCredential]::Empty
+        $NcRestCredential = [System.Management.Automation.PSCredential]::Empty,
+
+        [Parameter(Mandatory = $true, ParameterSetName = 'RestCertificate')]
```
Comment (luyaof) [inline src/modules/SdnDiag.Health/private/Test-NetworkInterfaceAPIDuplicateMacAddress.ps1:17]: Review `Mandatory` attribute

File: src/modules/SdnDiag.Health/private/Test-ResourceConfigurationState.ps1 [hunk 6]
```diff
@@ -4,24 +4,39 @@ function Test-ResourceConfigurationState {
...
         [SdnFabricEnvObject]$SdnEnvironmentObject,
 
-        [Parameter(Mandatory = $false)]
+        [Parameter(Mandatory = $false, ParameterSetName = 'RestCredential')]
         [System.Management.Automation.PSCredential]
         [System.Management.Automation.Credential()]
-        $NcRestCredential = [System.Management.Automation.PSCredential]::Empty
+        $NcRestCredential = [System.Management.Automation.PSCredential]::Empty,
+
+        [Parameter(Mandatory = $true, ParameterSetName = 'RestCertificate')]
```
Comment (luyaof) [inline src/modules/SdnDiag.Health/private/Test-ResourceConfigurationState.ps1:17]: Review `Mandatory` attribute

File: src/modules/SdnDiag.Health/private/Test-ResourceProvisioningState.ps1 [hunk 7]
```diff
@@ -4,24 +4,39 @@ function Test-ResourceProvisioningState {
...
         [SdnFabricEnvObject]$SdnEnvironmentObject,
 
-        [Parameter(Mandatory = $false)]
+        [Parameter(Mandatory = $false, ParameterSetName = 'RestCredential')]
         [System.Management.Automation.PSCredential]
         [System.Management.Automation.Credential()]
-        $NcRestCredential = [System.Management.Automation.PSCredential]::Empty
+        $NcRestCredential = [System.Management.Automation.PSCredential]::Empty,
+
+        [Parameter(Mandatory = $true, ParameterSetName = 'RestCertificate')]
```
Comment (luyaof) [inline src/modules/SdnDiag.Health/private/Test-ResourceProvisioningState.ps1:17]: Review `Mandatory` attribute

File: src/modules/SdnDiag.Health/private/Test-ServerHostId.ps1 [hunk 8]
```diff
@@ -14,12 +14,27 @@ function Test-ServerHostId {
...
         $Credential = [System.Management.Automation.PSCredential]::Empty,
 
-        [Parameter(Mandatory = $false)]
+        [Parameter(Mandatory = $false, ParameterSetName = 'RestCredential')]
         [System.Management.Automation.PSCredential]
         [System.Management.Automation.Credential()]
-        $NcRestCredential = [System.Management.Automation.PSCredential]::Empty
+        $NcRestCredential = [System.Management.Automation.PSCredential]::Empty,
+
+        [Parameter(Mandatory = $true, ParameterSetName = 'RestCertificate')]
```
Comment (luyaof) [inline src/modules/SdnDiag.Health/private/Test-ServerHostId.ps1:22]: Review `Mandatory` attribute

File: src/modules/SdnDiag.Health/private/Test-SlbManagerConnectionToMux.ps1 [hunk 9]
```diff
@@ -14,12 +14,27 @@ function Test-SlbManagerConnectionToMux {
...
         $Credential = [System.Management.Automation.PSCredential]::Empty,
 
-        [Parameter(Mandatory = $false)]
+        [Parameter(Mandatory = $false, ParameterSetName = 'RestCredential')]
         [System.Management.Automation.PSCredential]
         [System.Management.Automation.Credential()]
-        $NcRestCredential = [System.Management.Automation.PSCredential]::Empty
+        $NcRestCredential = [System.Management.Automation.PSCredential]::Empty,
+
+        [Parameter(Mandatory = $true, ParameterSetName = 'RestCertificate')]
```
Comment (luyaof) [inline src/modules/SdnDiag.Health/private/Test-SlbManagerConnectionToMux.ps1:22]: Review `Mandatory` attribute

File: src/modules/SdnDiag.LoadBalancerMux/public/Start-SdnMuxCertificateRotation.ps1 [hunk 10]
```diff
@@ -88,8 +96,33 @@ function Start-SdnMuxCertificateRotation {
...
+        Uri = $null
+        UseBasicParsing = $true
+    }
+    $confirmStateParams = @{
+        TimeoutInSec = 600
+        UseBasicParsing = $true
+    }
+
+    if ($PSBoundParameters.ContainsKey('NcRestCertificate')) {
+        $restCredParam = @{ NcRestCertificate = $NcRestCertificate }
```
Comment (luyaof) [inline src/modules/SdnDiag.LoadBalancerMux/public/Start-SdnMuxCertificateRotation.ps1:116]: Is it proper to merge the `$restCredParam` and `$ncRestParams`?

File: src/modules/SdnDiag.NetworkController/private/Get-SdnDiscovery.ps1 [hunk 11]
```diff
@@ -4,23 +4,42 @@ function Get-SdnDiscovery {
...
         [Uri]$NcUri,
 
-        [Parameter(Mandatory = $false)]
+        [Parameter(Mandatory = $false, ParameterSetName = 'RestCredential')]
         [System.Management.Automation.PSCredential]
         [System.Management.Automation.Credential()]
-        $Credential = [System.Management.Automation.PSCredential]::Empty
+        $NcRestCredential = [System.Management.Automation.PSCredential]::Empty,
+
+        [Parameter(Mandatory = $true, ParameterSetName = 'RestCertificate')]
```
Comment (luyaof) [inline src/modules/SdnDiag.NetworkController/private/Get-SdnDiscovery.ps1:24]: Review `Mandatory` attribute

File: src/modules/SdnDiag.NetworkController/private/Get-SdnVirtualServer.ps1 [hunk 12]
```diff
@@ -20,14 +24,30 @@ function Get-SdnVirtualServer {
...
         [switch]$ManagementAddressOnly,
 
-        [Parameter(Mandatory = $false)]
+        [Parameter(Mandatory = $false, ParameterSetName = 'RestCredential')]
         [System.Management.Automation.PSCredential]
         [System.Management.Automation.Credential()]
-        $Credential = [System.Management.Automation.PSCredential]::Empty
+        $NcRestCredential = [System.Management.Automation.PSCredential]::Empty,
+
+        [Parameter(Mandatory = $true, ParameterSetName = 'RestCertificate')]
```
Comment (luyaof) [inline src/modules/SdnDiag.NetworkController/private/Get-SdnVirtualServer.ps1:32]: Review `Mandatory` attribute

File: src/modules/SdnDiag.NetworkController/private/Invoke-SdnNetworkControllerStateDump.ps1 [hunk 13]
```diff
@@ -4,19 +4,27 @@ function Invoke-SdnNetworkControllerStateDump {
...
         [uri]$NcUri,
 
-        [Parameter(Mandatory = $false)]
+        [Parameter(Mandatory = $false, ParameterSetName = 'RestCredential')]
         [System.Management.Automation.PSCredential]
         [System.Management.Automation.Credential()]
-        $Credential = [System.Management.Automation.PSCredential]::Empty,
+        $NcRestCredential = [System.Management.Automation.PSCredential]::Empty,
+
+        [Parameter(Mandatory = $true, ParameterSetName = 'RestCertificate')]
```
Comment (luyaof) [inline src/modules/SdnDiag.NetworkController/private/Invoke-SdnNetworkControllerStateDump.ps1:26]: Review `Mandatory` attribute

File: src/modules/SdnDiag.NetworkController/public/Get-SdnAuditLog.ps1 [hunk 14]
```diff
@@ -7,27 +7,33 @@ function Get-SdnAuditLog {
...
         [Parameter(Mandatory = $true)]
         [Uri]$NcUri,
 
-        [Parameter(Mandatory = $false)]
+        [Parameter(Mandatory = $false, ParameterSetName = 'RestCredential')]
         [System.Management.Automation.PSCredential]
         [System.Management.Automation.Credential()]
         $NcRestCredential = [System.Management.Automation.PSCredential]::Empty,
 
+        [Parameter(Mandatory = $true, ParameterSetName = 'RestCertificate')]
```
Comment (luyaof) [inline src/modules/SdnDiag.NetworkController/public/Get-SdnAuditLog.ps1:34]: Review `Mandatory` attribute

File: src/modules/SdnDiag.NetworkController/public/Get-SdnInfrastructureInfo.ps1 [hunk 15]
```diff
@@ -37,7 +39,10 @@ function Get-SdnInfrastructureInfo {
         })]
         [Uri]$NcUri,
 
-        [Parameter(Mandatory = $false)]
+        [Parameter(Mandatory = $true, ParameterSetName = 'RestCertificate')]
```
Comment (luyaof) [inline src/modules/SdnDiag.NetworkController/public/Get-SdnInfrastructureInfo.ps1:42]: Review `Mandatory` attribute

File: src/modules/SdnDiag.NetworkController/public/Get-SdnNetworkControllerState.ps1 [hunk 16]
```diff
@@ -29,14 +32,32 @@ function Get-SdnNetworkControllerState {
         [System.Management.Automation.Credential()]
         $Credential = [System.Management.Automation.PSCredential]::Empty,
 
-        [Parameter(Mandatory = $false)]
+        [Parameter(Mandatory = $false, ParameterSetName = 'RestCredential')]
         [System.Management.Automation.PSCredential]
         [System.Management.Automation.Credential()]
         $NcRestCredential = [System.Management.Automation.PSCredential]::Empty,
 
+        [Parameter(Mandatory = $true, ParameterSetName = 'RestCertificate')]
```
Comment (luyaof) [inline src/modules/SdnDiag.NetworkController/public/Get-SdnNetworkControllerState.ps1:40]: Review `Mandatory` attribute

File: src/modules/SdnDiag.NetworkController/public/Get-SdnNetworkInterfaceOutboundPublicIPAddress.ps1 [hunk 17]
```diff
@@ -6,39 +6,58 @@ function Get-SdnNetworkInterfaceOutboundPublicIPAddress {
...
         [System.String]$ResourceId,
 
-        [Parameter(Mandatory = $false)]
+        [Parameter(Mandatory = $false, ParameterSetName = 'RestCredential')]
         [System.Management.Automation.PSCredential]
         [System.Management.Automation.Credential()]
-        $Credential = [System.Management.Automation.PSCredential]::Empty
+        $NcRestCredential = [System.Management.Automation.PSCredential]::Empty,
+
+        [Parameter(Mandatory = $true, ParameterSetName = 'RestCertificate')]
```
Comment (luyaof) [inline src/modules/SdnDiag.NetworkController/public/Get-SdnNetworkInterfaceOutboundPublicIPAddress.ps1:33]: Review `Mandatory` attribute

File: src/modules/SdnDiag.NetworkController/public/Get-SdnPublicIPPoolUsageSummary.ps1 [hunk 18]
```diff
@@ -7,27 +7,44 @@ function Get-SdnPublicIPPoolUsageSummary {
...
         [Uri]$NcUri,
 
-        [Parameter(Mandatory = $false)]
+        [Parameter(Mandatory = $false, ParameterSetName = 'RestCredential')]
         [System.Management.Automation.PSCredential]
         [System.Management.Automation.Credential()]
-        $NcRestCredential = [System.Management.Automation.PSCredential]::Empty
+        $NcRestCredential = [System.Management.Automation.PSCredential]::Empty,
+
+        [Parameter(Mandatory = $true, ParameterSetName = 'RestCertificate')]
```
Comment (luyaof) [inline src/modules/SdnDiag.NetworkController/public/Get-SdnPublicIPPoolUsageSummary.ps1:27]: Review `Mandatory` attribute

File: src/modules/SdnDiag.NetworkController/public/Get-SdnSlbStateInformation.ps1 [hunk 19]
```diff
@@ -17,23 +20,26 @@ function Get-SdnSlbStateInformation {
...
         [IPAddress]$VirtualIPAddress,
 
-        [Parameter(Mandatory = $false)]
+        [Parameter(Mandatory = $false, ParameterSetName = 'RestCredential')]
         [System.Management.Automation.PSCredential]
         [System.Management.Automation.Credential()]
-        $Credential = [System.Management.Automation.PSCredential]::Empty,
+        $NcRestCredential = [System.Management.Automation.PSCredential]::Empty,
+
+        [Parameter(Mandatory = $true, ParameterSetName = 'RestCertificate')]
```
Comment (luyaof) [inline src/modules/SdnDiag.NetworkController/public/Get-SdnSlbStateInformation.ps1:41]: Review `Mandatory` attribute

File: src/modules/SdnDiag.NetworkController/public/Invoke-SdnResourceDump.ps1 [hunk 20]
```diff
@@ -4,38 +4,55 @@ function Invoke-SdnResourceDump {
...
+    [CmdletBinding(DefaultParameterSetName = 'RestCredential')]
     param (
         [Parameter(Mandatory = $true)]
         [Uri]$NcUri,
 
         [Parameter(Mandatory = $true)]
         [System.IO.FileInfo]$OutputDirectory,
 
-        [Parameter(Mandatory = $false)]
+        [Parameter(Mandatory = $true, ParameterSetName = 'RestCertificate')]
```
Comment (luyaof) [inline src/modules/SdnDiag.NetworkController/public/Invoke-SdnResourceDump.ps1:27]: Review `Mandatory` attribute

File: src/modules/SdnDiag.Utilities/private/Invoke-RestMethodWithRetry.ps1 [hunk 21]
```diff
@@ -21,7 +23,10 @@ function Invoke-RestMethodWithRetry {
         [Parameter(Mandatory = $false)]
         [Switch] $UseBasicParsing,
 
-        [Parameter(Mandatory = $false)]
+        [Parameter(Mandatory = $true, ParameterSetName = 'Certificate')]
```
Comment (luyaof) [inline src/modules/SdnDiag.Utilities/private/Invoke-RestMethodWithRetry.ps1:26]: Review `Mandatory` attribute

File: src/modules/SdnDiag.Utilities/private/Invoke-WebRequestWithRetry.ps1 [hunk 22]
```diff
@@ -21,20 +23,23 @@ function Invoke-WebRequestWithRetry {
         [Parameter(Mandatory = $false)]
         [Switch] $UseBasicParsing,
 
-        [Parameter(Mandatory = $false)]
+        [Parameter(Mandatory = $true, ParameterSetName = 'Certificate')]
```
Comment (luyaof) [inline src/modules/SdnDiag.Utilities/private/Invoke-WebRequestWithRetry.ps1:26]: Review `Mandatory` attribute

**Review by luyaof — CHANGES_REQUESTED** [review]
Several places have `NcRestCertificate` parameter set as mandatory, need to be reviewed

File: src/modules/SdnDiag.Health/private/Test-NcHostAgentConnectionToApiService.ps1 (hunk 3 above)
Comment (arudell) [reply src/modules/SdnDiag.Health/private/Test-NcHostAgentConnectionToApiService.ps1:22]: This is correct when using ParameterSetName and helps define what is required. It enables unique properties to be required depending on the ParameterSetName that is used.

File: src/modules/SdnDiag.Health/private/Test-NetworkControllerCertCredential.ps1 (hunk 4 above)
Comment (arudell) [reply src/modules/SdnDiag.Health/private/Test-NetworkControllerCertCredential.ps1:22]: This is correct when using ParameterSetName and helps define what is required. It enables unique properties to be required depending on the ParameterSetName that is used.

File: src/modules/SdnDiag.Health/private/Test-NetworkInterfaceAPIDuplicateMacAddress.ps1 (hunk 5 above)
Comment (arudell) [reply src/modules/SdnDiag.Health/private/Test-NetworkInterfaceAPIDuplicateMacAddress.ps1:17]: This is correct when using ParameterSetName and helps define what is required. It enables unique properties to be required depending on the ParameterSetName that is used.

File: src/modules/SdnDiag.NetworkController/public/Get-SdnPublicIPPoolUsageSummary.ps1 (hunk 18 above)
Comment (arudell) [reply src/modules/SdnDiag.NetworkController/public/Get-SdnPublicIPPoolUsageSummary.ps1:27]: This is correct when using ParameterSetName and helps define what is required. It enables unique properties to be required depending on the ParameterSetName that is used.

File: src/modules/SdnDiag.NetworkController/public/Get-SdnSlbStateInformation.ps1 (hunk 19 above)
Comment (arudell) [reply src/modules/SdnDiag.NetworkController/public/Get-SdnSlbStateInformation.ps1:41]: This is correct when using ParameterSetName and helps define what is required. It enables unique properties to be required depending on the ParameterSetName that is used.

File: src/modules/SdnDiag.Utilities/private/Invoke-WebRequestWithRetry.ps1 (hunk 22 above)
Comment (arudell) [reply src/modules/SdnDiag.Utilities/private/Invoke-WebRequestWithRetry.ps1:26]: This is correct when using ParameterSetName and helps define what is required. It enables unique properties to be required depending on the ParameterSetName that is used.

File: src/modules/SdnDiag.Health/private/Test-ResourceConfigurationState.ps1 (hunk 6 above)
Comment (arudell) [reply src/modules/SdnDiag.Health/private/Test-ResourceConfigurationState.ps1:17]: This is correct when using ParameterSetName and helps define what is required. It enables unique properties to be required depending on the ParameterSetName that is used.

File: src/modules/SdnDiag.Health/private/Test-ResourceProvisioningState.ps1 (hunk 7 above)
Comment (arudell) [reply src/modules/SdnDiag.Health/private/Test-ResourceProvisioningState.ps1:17]: This is correct when using ParameterSetName and helps define what is required. It enables unique properties to be required depending on the ParameterSetName that is used.

File: src/modules/SdnDiag.Health/private/Test-SlbManagerConnectionToMux.ps1 (hunk 9 above)
Comment (arudell) [reply src/modules/SdnDiag.Health/private/Test-SlbManagerConnectionToMux.ps1:22]: This is correct when using ParameterSetName and helps define what is required. It enables unique properties to be required depending on the ParameterSetName that is used.

File: src/modules/SdnDiag.NetworkController/private/Get-SdnDiscovery.ps1 (hunk 11 above)
Comment (arudell) [reply src/modules/SdnDiag.NetworkController/private/Get-SdnDiscovery.ps1:24]: This is correct when using ParameterSetName and helps define what is required. It enables unique properties to be required depending on the ParameterSetName that is used.

File: src/modules/SdnDiag.NetworkController/public/Get-SdnInfrastructureInfo.ps1 (hunk 15 above)
Comment (arudell) [reply src/modules/SdnDiag.NetworkController/public/Get-SdnInfrastructureInfo.ps1:42]: This is correct when using ParameterSetName and helps define what is required. It enables unique properties to be required depending on the ParameterSetName that is used.

File: src/modules/SdnDiag.NetworkController/public/Get-SdnNetworkInterfaceOutboundPublicIPAddress.ps1 (hunk 17 above)
Comment (arudell) [reply src/modules/SdnDiag.NetworkController/public/Get-SdnNetworkInterfaceOutboundPublicIPAddress.ps1:33]: This is correct when using ParameterSetName and helps define what is required. It enables unique properties to be required depending on the ParameterSetName that is used.

File: src/modules/SdnDiag.NetworkController/public/Get-SdnNetworkControllerState.ps1 (hunk 16 above)
Comment (arudell) [reply src/modules/SdnDiag.NetworkController/public/Get-SdnNetworkControllerState.ps1:40]: This is correct when using ParameterSetName and helps define what is required. It enables unique properties to be required depending on the ParameterSetName that is used.

File: src/modules/SdnDiag.NetworkController/public/Get-SdnAuditLog.ps1 (hunk 14 above)
Comment (arudell) [reply src/modules/SdnDiag.NetworkController/public/Get-SdnAuditLog.ps1:34]: This is correct when using ParameterSetName and helps define what is required. It enables unique properties to be required depending on the ParameterSetName that is used.

File: src/modules/SdnDiag.NetworkController/private/Get-SdnVirtualServer.ps1 (hunk 12 above)
Comment (arudell) [reply src/modules/SdnDiag.NetworkController/private/Get-SdnVirtualServer.ps1:32]: This is correct when using ParameterSetName and helps define what is required. It enables unique properties to be required depending on the ParameterSetName that is used.

File: src/modules/SdnDiag.Health/private/Test-ServerHostId.ps1 (hunk 8 above)
Comment (arudell) [reply src/modules/SdnDiag.Health/private/Test-ServerHostId.ps1:22]: This is correct when using ParameterSetName and helps define what is required. It enables unique properties to be required depending on the ParameterSetName that is used.

**Review by luyaof — APPROVED** [review]

**Pull request merged**